lib/elastic_apm/timed_worker.rb

```diff
@@ -50,21 +50,25 @@ def run_once
 
     private
 
+    # rubocop:disable Metrics/MethodLength
     def process_messages
+      should_exit = false
+
       while (msg = messages.pop(true))
```

Comment:
Giving my previous comment a second thought, something similar could happen here: E.g a bad deploy happens that causes tons of errors to be sent, the `while` loop would go on for quite some time, blocking transactions to be sent, as the thread gets stuck in processing the enqueued `messages`. 

With the changes that you are proposing in this PR the error messages are not directly sent when they occur anymore, but eventually when the thread processes the queue. With this change, have you considered to send all the currently queued errors at once, instead of one by one?

Reply:
So essentially doing the same for errors as for transactions?

Reply:
basically yes, but with a `should_flush_errors?` check that returns `true` as soon as an error is enqueued.